src/net_setup.c

```diff
@@ -564,12 +580,16 @@ bool setup_myself_reloadable(void) {
 			routing_mode = RMODE_HUB;
 		} else {
 			logger(DEBUG_ALWAYS, LOG_ERR, "Invalid routing mode!");
+			free(rmode);
 			return false;
 		}
 
 		free(rmode);
+		rmode = NULL;
```

Comment:
This seems unnecessary, `rmode` is just a local variable that's no longer used after this. If we want to make this more explicit, we could put braces around each block that checks a certain config variable, like so:

```
{
    char *rmode = NULL;

    if(get_config_string(lookup_config(config_tree, "Mode"), &rmode)) {
        ...
    }
}
```